neurom/features/morphology.py

```diff
@@ -428,27 +432,28 @@ def _mean_radius(neurite):
                 return points[~valid_max, COLS.R][0]
         return points[valid_pts, COLS.R].mean()
 
-    return [_mean_radius(n) for n in iter_neurites(morph, filt=is_type(neurite_type))]
+    return _map_neurites(_mean_radius, morph, neurite_type)
 
 
 @feature(shape=(...,))
 def trunk_section_lengths(morph, neurite_type=NeuriteType.all):
     """List of lengths of trunk sections of neurites in a morph."""
-    return [morphmath.section_length(n.root_node.points)
-            for n in iter_neurites(morph, filt=is_type(neurite_type))]
+    def trunk_section_length(neurite):
+        return morphmath.section_length(neurite.root_node.points)
+
+    return _map_neurites(trunk_section_length, morph, neurite_type)
 
 
 @feature(shape=())
 def number_of_neurites(morph, neurite_type=NeuriteType.all):
     """Number of neurites in a morph."""
-    return sum(1 for _ in iter_neurites(morph, filt=is_type(neurite_type)))
+    return sum(_map_neurites(lambda _: 1, morph, neurite_type))
```

Comment:
FYI, I think the old way was done to reduce the list creation, so only an iterable is needed.  I don't think that perf is critical here, so for uniformity sake, I think it's ok; but something to keep in mind.

Reply:
The reason behind offloading almost everything to `_map_neurites` is mainly because it will be necessary for the mixed type subtree processing if we want to avoid adding extra logic in each feature.

It might be worth rethinking however at which stage of the feature calculation the type of the iterable (e.g. list) is applied.

Reply:
I could however replace `sum` with `len` so that we avoid the extra iteration.